src/io/iter_mnist.cc

```diff
@@ -47,6 +51,10 @@ struct MNISTParam : public dmlc::Parameter<MNISTParam> {
         .describe("Augmentation Param: Random Seed.");
     DMLC_DECLARE_FIELD(silent).set_default(false)
         .describe("Auxiliary Param: Whether to print out data info.");
+    DMLC_DECLARE_FIELD(num_parts).set_default(1)
```

Comment:
Maybe no need for MNIST iter. Since we don't need to distributed it, do it for ImageRecord